lib/explorer/polars_backend/expression.ex

```diff
@@ -0,0 +1,40 @@
+defmodule Explorer.PolarsBackend.Expression do
+  @moduledoc false
+  # This module is responsible for translating the opaque LazySeries
+  # to polars expressions in the Rust side.
+
+  alias Explorer.DataFrame
+  alias Explorer.Backend.LazySeries
+  alias Explorer.PolarsBackend.Native
+
+  defstruct resource: nil, reference: nil
+
+  @type t :: %__MODULE__{resource: binary(), reference: reference()}
+
+  # We are going to generate all functions for each valid operation.
+  for {op, arity} <- Explorer.Backend.LazySeries.operations() do
+    args = Macro.generate_arguments(arity, __MODULE__)
+
+    updates =
+      for arg <- args do
+        quote do
+          to_expr(unquote(arg))
+        end
+      end
+
+    expr_op = :"expr_#{op}"
+
+    def to_expr(%LazySeries{op: unquote(op), args: unquote(args)}) do
+      Native.unquote(expr_op)(unquote_splicing(updates))
+    end
+  end
+
+  def to_expr(number) when is_integer(number), do: Native.expr_integer(number)
+  def to_expr(number) when is_float(number), do: Native.expr_float(number)
+  def to_expr(binary) when is_binary(binary), do: binary
```

Comment:
We should, but `column` only accepts a raw string, so I kept this for simplicity. I'm going to change in the next PR :)

Reply:
Cool! I couldn't use `definitions_in/2` because it uses the name of the function (to_expr). But I added it manually https://github.com/elixir-nx/explorer/pull/290/files#diff-d28abb6325b1243b94eb349b2f2b8c22b9f775ce5cf5710f4589d4ebf4f2f415R20